pages/missions.php

```diff
@@ -168,7 +168,14 @@ function runActiveMission() {
                         else if($player->mission_stage['stage_id'] > 2){
                             $player->mission_stage['stage_id'] -= 1;
                         }
-                        $mission_status = $mission->nextStage($player->mission_stage['stage_id']);
+                        if ($player->location == $player->village_location)
+                        {
+                            $mission_status = $mission->nextStage($player->mission_stage['stage_id'] = 4);
+                        }
+                        else
+                        {
+                            $mission_status = $mission->nextStage($player->mission_stage['stage_id']);
+                        }
```

Comment:
```suggestion
                        if ($player->location == $player->village_location) {
                            $player->mission_stage['stage_id'] = 4;
                        }  
                        
                        $mission_status = $mission->nextStage($player->mission_stage['stage_id']);
```

I think this way might be a little cleaner to read